src/main/java/com/palantir/gradle/shadowjar/JarFilesRelocator.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.shadowjar;
+
+import com.github.jengelman.gradle.plugins.shadow.relocation.CacheableRelocator;
+import com.github.jengelman.gradle.plugins.shadow.relocation.RelocateClassContext;
+import com.github.jengelman.gradle.plugins.shadow.relocation.RelocatePathContext;
+import com.github.jengelman.gradle.plugins.shadow.relocation.SimpleRelocator;
+import com.google.common.base.Suppliers;
+import com.google.common.collect.ImmutableList;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Supplier;
+import org.gradle.api.provider.Provider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@CacheableRelocator
```

Comment:
should we run the tests with `--build-cache` to make sure the second run will use the caches ?